<a href="https://colab.research.google.com/github/H2D2R2/Dz3/blob/main/%D0%94%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D0%B5%D0%B5_%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import warnings
from math import factorial
from math import sqrt

# Задание 1

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [ ]:
zp = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

In [ ]:
s = sum (zp)  / len (zp)
s

65.3

In [ ]:
sk = sqrt(((len (zp) - s) **2) / len (zp))
sk

10.129387938074046

In [ ]:
sk1 = sqrt(((len (zp) - s) **2) / (len (zp) -1))
sk1

10.392532744336448

Ответ: 
Среднее арифметическое - 65.3;среднее квадратичное отклонение - 10.1; смещенная оценка дисперсии - 10.1; несмещенная оценка дисперсии - 10.3;  


# Задание 2

В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [ ]:
def comb(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

p1 = (comb (5,2) / comb (8,2))  *  ((comb (5,1) * comb(7,3)) / comb (12,4))
p2 = ((comb (5,1) * comb (3,1)) / comb (8,2)) * ((comb(5,2) * comb (7,2)) / comb (12,4))
p3 = (comb (3,2) / comb (8,2)) * (comb (5,3)/ comb(12,4))

(p1+p2+p3) *100

35.56998556998557

Ответ: 35.5%

# Задание 3

На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

А) 

In [ ]:
#решаем по формуле Байенса

B1 = 0.9
B2 = 0.8
B3 = 0.6

A = (1/3 * B1) + (1/3 * B2) + (1/3 * B3)

(B1 * 1/3) / A *100

39.130434782608695

Б)

In [ ]:
(B2 * 1/3) / A *100

34.78260869565218

В)


In [ ]:
(B3 * 1/3) / A *100

26.08695652173913

Ответ: A) 39.1%; Б) 34.1%; В) 26%;

# Задание 4

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

In [ ]:
#решаем по формуле Байенса

B1 = 0.8
B2 = 0.7
B3 = 0.9

A = (1/2 * B1) + (1/4 * B2) + (1/4 * B3)

(B1 * 1/3) / A *100

33.333333333333336

In [ ]:
(B2 * 1/3) / A *100

29.166666666666668

In [ ]:
(B3 * 1/3) / A *100

37.5

Ответ: A) 33.3%; Б) 29.16%; В) 37.5%;

# Задание 5 

Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

In [ ]:
a = 0.1
b = 0.2
c = 0.25

In [ ]:
d= a * b * c                                 #все детали
d

0.005000000000000001

In [ ]:
v = (a * b) + (a * c) + (b * c)             #только две детали
v

0.095

In [ ]:
(a+b+c) + v + d                            #хотя бы одна деталь (понимаю, что не верно. Не понимаю, почему)

0.65

In [ ]:
(a+b+c) + v                                #от одной до двух деталей (тоже не верно, тоже не понимаю почему)

0.645